### PR TITLE
Partially fixes #5981

### LINKED ISSUE #5981: Upgrader - CLI errors
#### Description
There are several errors reported during an upgrade via CLI that are hidden when upgrading via the browser (e.g, eval'd code).  These errors does not happen on a 2.1=>2.1 upgrade, you must be upgrading from 1.x or 2.0.  

image

These reported errors do not seem to cause failures, but should be cleaned up nonetheless.

### Steps to reproduce
1. Upgrade, via CLI, from a valid 2.0 DB

### Environment (complete as necessary)
- Version/Git revision:  Current
- Database Type: mysql
- Database Version: 5.7
- PHP Version: 7.3

### PR DESCRIPTION
**Pull request by jdarwood007**
Does not yet fix the convert_utf8 template issue

### CODE REVIEW

**Comment by sbulen** [comment]
I'll do some tests of this once #5961 has been merged...

**Comment by sbulen** [comment]
This is what I am seeing as of now:
image

**Comment by Sesquipedalian** [comment]
The errors on line 4923 could be solved simply by changing `if (is_null($arIp[0]))` to `if (!isset($arIp[0]))`.

File: other/upgrade.php [hunk 1]
```diff
@@ -4946,7 +4946,7 @@ function MySQLConvertOldIp($targetTable, $oldCol, $newCol, $limit = 50000, $setS
 		$smcFunc['db_free_result']($request);
 
 		// Special case, null ip could keep us in a loop.
-		if (is_null($arIp[0]))
+		if (!isset($arIp[0]) || is_null($arIp[0]) || $arIp[0] == '')
```
Comment (Sesquipedalian) [inline other/upgrade.php]: `!isset($arIp[0]) || is_null($arIp[0])` is redundant.

**Review by Sesquipedalian — CHANGES_REQUESTED** [review]

File: other/upgrade.php (hunk 1 above)
Comment (jdarwood007) [reply other/upgrade.php]: So was the other check.

**Comment by sbulen** [comment]
If you merge this (& keep #5981 open) I'll look into the other issues.

File: other/upgrade.php (hunk 1 above)
Comment (sbulen) [reply other/upgrade.php]: This works as-is.   empty() oughtta work here, too, covers all the bases, I think...

**Comment by jdarwood007** [comment]
I'm not sure why, but I'm thinking empty would create a issue.  My thoughts are thinking it has something to do with 0s but I can't recall exactly.

**Comment by sbulen** [comment]
I've been testing this extensively along with all the other upgrader changes & it works just fine, I suggest merging it.

**Comment by sbulen** [comment]
Since some of these issues impact current upgrades, I think we should merge this one asap.

**Review by MissAllSunday — APPROVED** [review]

**Pull request merged**